lms/djangoapps/grades/new/course_data.py

```diff
@@ -82,7 +83,7 @@ def version(self):
     @property
     def edited_on(self):
         # get course block from structure only; subtree_edited_on field on modulestore's course block isn't optimized.
-        course_block = self.structure[self.location]
+        course_block = self.structure[self.location] if self.structure else {}
```

Comment:
I also had to add this to avoid the following test failure:

```
Traceback (most recent call last):
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mock.py", line 1618, in _inner
    return f(*args, **kw)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/tests/test_view_authentication.py", line 356, in test_dark_launch_instructor
    self._check_staff(self.course)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/tests/test_view_authentication.py", line 100, in _check_staff
    self.assert_request_status_code(404, url)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/tests/helpers.py", line 45, in assert_request_status_code
    response = make_request(url, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/test/client.py", line 500, in get
    **extra)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/test/client.py", line 303, in get
    return self.generic('GET', path, secure=secure, **r)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/test/client.py", line 379, in generic
    return self.request(**r)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/test/client.py", line 466, in request
    six.reraise(*exc_info)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/contrib/auth/decorators.py", line 22, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/views/decorators/cache.py", line 43, in _cache_controlled
    response = viewfunc(request, *args, **kw)
  File "/edx/app/edxapp/edx-platform/common/djangoapps/util/views.py", line 51, in inner
    response = view_func(request, *args, **kwargs)
  File "/edx/app/edxapp/edx-platform/openedx/features/enterprise_support/api.py", line 220, in inner
    return view_func(request, course_id, *args, **kwargs)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/views/views.py", line 816, in progress
    return _progress(request, course_key, student_id)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/views/views.py", line 875, in _progress
    course_grade = CourseGradeFactory().create(student, course)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/new/course_grade_factory.py", line 46, in create
    return self._update(user, course_data, read_only)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/new/course_grade_factory.py", line 183, in _update
    course_data.full_string(), user.id, course_grade, should_persist,
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/new/course_data.py", line 98, in full_string
    self.course_key, self.version, self.edited_on, self.grading_policy_hash,
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/new/course_data.py", line 86, in edited_on
    course_block = self.structure[self.location]
  File "/edx/app/edxapp/edx-platform/openedx/core/djangoapps/content/block_structure/block_structure.py", line 448, in __getitem__
    return self._block_data_map[usage_key]
KeyError: BlockUsageLocator(CourseLocator(u'org.11', u'999', u'Robot_Super_Course', None, None), u'course', u'Robot_Super_Course')
```